client/src/partials/patient_edit/edit_patient.js

```diff
@@ -189,9 +189,23 @@ angular.module('bhima.controllers')
       customValidation();
     });
 
+    // Restart the search
+    $scope.restartSearch = function () {
+      originalPatientData = null;
+      $scope.patient = { origin_location_id: null, current_location_id: null };
+      $scope.initialOriginLocation = null;
+      $scope.initialCurrentLocation = null;
+
+      // Make sure the patient query gets processed
+      if ('processed' in dependencies.patient) {
```

Comment:
Unfortunately, this does not quite do what is needed.  For the first time a patient is selected, it should process both 'patient' and 'debtorGroup'.  In subsequent calls, it only needs to execute the 'patient' query.    Since there is no need to process the 'debtorGroup' again, it seems like the first time it should do both, but subsequent times only 'patient'.  I don't think adding logic for that is very clean.

I cannot do the valdiate.refresh() call in the restartSearch() function becuase validate.refresh() also calls 'validate.process()' which simply looks up the old data again.   I suppose I could add a function to 'validate' that would clear the selected 'processed' flags.    Suggestions?